backend.py

```diff
@@ -447,33 +459,25 @@ def getAttendance(self) -> List:
 
         attendanceData = self.wclg.getAttendance(token)
 
-        attendance = {}
-        raids = []
-        formattedraids = []
+        attendance = collections.defaultdict(dict)
+        log_amts = collections.defaultdict(int)
         for raid in attendanceData:
-            raids.append((raid["code"], raid["startTime"]))
-            for player in raid["players"]:
-                if player["name"] not in attendance and len(raids)>0:
-                    tmp = {}
-                    for r in raids[:-1]:
-                        tmp[r[0]] = 0
-                    attendance[player["name"]] = tmp
-                attendance[player["name"]][raid["code"]] = str(player["presence"])
-
-                
-
-        for raid in raids:
-            raidcode = raid[0]
-            for player in attendance:
-                if raidcode not in attendance[player]:
-                    attendance[player][raidcode] = "0"
-
-        for entry in raids:
-            new1 = time.strftime("%d/%m", time.gmtime(entry[1]))
-            formattedraids.append((entry[0],new1))
-
-        print(formattedraids)
-        return formattedraids, attendance, token
+            players = [p['name'] for p in raid['players']]
+            raid_id = raid['zone']['id']
+            presences = [p['presence'] for p in raid['players']]
+            raid_att = attendance[raid_id]
+            num_logs = log_amts[raid_id]
+            for player, presence in zip(players, presences):
+                if player not in raid_att:
+                    raid_att[player] = [0] * num_logs
+                raid_att[player].append(presence)
+            attendance[raid_id] = raid_att
+            log_amts[raid_id] += 1
+            for player in raid_att:
+                vals_missing = log_amts[raid_id] - len(raid_att[player])
+                raid_att[player] += [0]*vals_missing
+
```

Comment:
Could add a loop that calculates attendance% from entries, since it's probably clearer to do here than in app.py. Could also just leave it to client side? It's not particularly expensive to calculate, a client side JS could do it pretty well

Reply:
Added a 4th return parameter containing attendance % for each raid for each player as a dict.